src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -153,6 +154,62 @@ public async Task<HttpResponseMessage> HandleRequestAsync(HttpRequestMessage req
                     return await this.HandleStartOrchestratorRequestAsync(request, functionName, instanceId);
                 }
 
+                i = path.IndexOf(EntitiesControllerSegment, StringComparison.OrdinalIgnoreCase);
+                if (i >= 0 && (request.Method == HttpMethod.Get || request.Method == HttpMethod.Post))
+                {
+                    EntityId entityId;
+
+                    i += EntitiesControllerSegment.Length;
+                    nextSlash = path.IndexOf('/', i);
+
+                    try
+                    {
+                        if (nextSlash < 0)
+                        {
+                            entityId = new EntityId(path.Substring(i), string.Empty);
+                        }
+                        else
+                        {
+                            entityId = new EntityId(path.Substring(i, nextSlash - i), path.Substring(nextSlash + 1));
+                        }
+                    }
+                    catch (ArgumentException e)
+                    {
+                        return request.CreateErrorResponse(HttpStatusCode.BadRequest, e.Message);
+                    }
+
+                    if (request.Method == HttpMethod.Get)
+                    {
+                        return await this.HandleGetEntityRequestAsync(request, entityId);
+                    }
+                    else
+                    {
+                        return await this.HandlePostEntityOperationRequestAsync(request, entityId);
+                    }
+                }
+
+                if (i >= 0 && request.Method == HttpMethod.Post)

```

Comment:
Will this condition ever be `true`? If the condition at line 158 is true, then this method has already returned. If that condition is false, then `i` is still -1 or request.Method is not `Post`.

Reply:
Good catch - I must have accidentally duplicated the first paragraph of the function while editing.

This just goes to confirm that we should refactor this function.